src/main/java/org/truffleruby/interop/RubyMessageResolution.java

```diff
@@ -266,26 +264,20 @@ private String toString(Object name) {
             return name.toString();
         }
 
+        // TODO CS 22-Mar-18 this should be cached
+
+        @TruffleBoundary
+        private boolean isIVar(String name) {
+            return name.startsWith("@");
```

Comment:
`!name.isEmpty() && name.charAt(0) == '@'` might avoid the need to cache this. It should at least let you get rid of the boundary, if desirable.

Reply:
Fixing.